src/eventloop.c

```diff
@@ -241,45 +161,69 @@ void handle_input(uv_poll_t* handle, int status, int events) {
 */
 guint input_add(int fd, PurpleInputCondition cond,
                 PurpleInputFunction func, gpointer user_data) {
+    g_return_if_fail(cond == 1 || cond == 2);
+    g_return_if_fail(fd > 0);
+
+    /**
+     * There is some subtle logic to this function. LibUV can only handle
+     * one poll handle per FD, but libpurple wants us to be able to support
+     * multiple per FD. This means we need to manage multiple inputs for
+     * a single handle ourselves.
+     */
 
-    uv_poll_t *poll_handle = NULL;
-    uint32_t *pollsOpen;
+    // Create a struct to hold THIS event/func pair
+    s_evLoopInputEvent *input_event = g_malloc(sizeof(uv_poll_t));
```

Comment:
I don't think you should gamble on these having the same size ;)

Reply:
Oops!